example/forest-example.cpp

```diff
@@ -1,12 +1,67 @@
 #include <forest/forest.hpp>
+#include <cstdio>
+#include <functional>
 #include <iostream>
+#include <iterator>
 
 constexpr auto foo = forest::literal<64>("<rgb=500><b><i>hello</b></i></rgb> world");
 
+void with_temp_file(std::function<void(std::FILE*)> const fn) {
+	constexpr auto tf_name = "temp.txt";
+
+	auto temp_file = std::fopen(tf_name, "w+");
+
+	if (temp_file == nullptr) { return; }
+
+	fn(temp_file);
+
+	std::fclose(temp_file);
```

Comment:
This is currently a bit unsafe: eg, if `fn` throws, the file will remain locked and undeleted. I'd suggest wrapping the "close" logic into a lambda, and `temp_file` into `std::unique_ptr<std::FILE*, decltype(close_file)>`. That way the file will always be closed on destruction of the unique ptr / the function returning (even if it unwinds due to a throw).